packages/client/src/utils/gif.ts

```diff
@@ -0,0 +1,167 @@
+import { Frame, GifReader } from 'omggif'
+import axios, { isAxiosError } from 'axios'
+
+interface IGif {
+  url: string
+  loop?: boolean
+}
+
+interface IFrame extends Frame {
+  pixels: Uint8ClampedArray
+}
+
+export class Gif {
+  private _url: string
+  private _loop: boolean
+  private _loopCount: number
+  private _reader: GifReader | null
+  private _frames: IFrame[] = []
+  private _currentFrame: number
+  private _running: boolean
+  private _delayCompensation: number
+  private _lastTime: number
+  private _ctx: CanvasRenderingContext2D | null
+  public image: HTMLCanvasElement
+  public width: number
+  public height: number
+
+  constructor({ url, loop = true }: IGif) {
+    this._url = url
+    this._loop = loop
+    this._reader = null
+    this._currentFrame = 0
+    this._lastTime = 0
+    this._loopCount = 0
+    this._delayCompensation = 0
+    this._running = false
+    this.image = document.createElement('canvas')
+    this._ctx = this.image.getContext('2d')
+    this.width = 0
+    this.height = 0
+
+    this._init()
+  }
+
+  private _generateFrames(): void {
+    if (this._reader) {
+      this._frames = [...Array(this._reader.numFrames()).keys()].map(
+        (_, index) => {
+          const frame = this._reader?.frameInfo(index) as IFrame
+          const width = this._reader?.width
+          const height = this._reader?.height
+
+          if (width && height) {
+            frame.pixels = new Uint8ClampedArray(width * height * 4)
+          }
+
+          this._reader?.decodeAndBlitFrameRGBA(index, frame.pixels)
+          return frame
+        }
+      )
+    }
```

Comment:
Чтобы избежать [лишний уровень вложенности](https://learn.javascript.ru/coding-style#urovni-vlozhennosti) и увеличить читабельность предлагаю так:

```suggestion
    if (!this._reader) {
        return;
    }
    
    this._frames = [...Array(this._reader.numFrames()).keys()].map(
        (_, index) => {
          const frame = this._reader?.frameInfo(index) as IFrame
          const width = this._reader?.width
          const height = this._reader?.height

          if (width && height) {
            frame.pixels = new Uint8ClampedArray(width * height * 4)
          }

          this._reader?.decodeAndBlitFrameRGBA(index, frame.pixels)
          return frame
        }
      )
```

Reply:
готово